### ISSUE TITLE
workflow/UI tweak - upload page..

### ISSUE THREAD

**Issue by dobkeratops** [bbernhard/imagemonkey-core#231]
<img width="978" alt="Screenshot 2019-04-14 at 12 06 29" src="https://user-images.githubusercontent.com/1120754/56091964-d789db80-5ead-11e9-9ef4-11342406d920.png">

as the label list grows (which IMO it must ), this will get harder to use ;
My suggestion is for this to default to 'anything' , so someone can safely submit an image instantly... no barrier to casual contribution;

you could rely more on the **add labels** UI for people to classify - the fact you have a common label selection there is great (that could be dynamic, e.g if an image has no labels, you start with broad categories & scene labels, and if the image has a scene label you could show the common objects for that scene) . If your task list was ever empty, you could default to showing people that page

**Comment by bbernhard** [comment]
you are totally right, the label selection in the "donate view" gets harder and harder the more labels we add. I would go even further and completely remove the label dropdown from the donate page and always default to < unlabeled >, see also my last post here: https://github.com/bbernhard/imagemonkey-core/issues/18

**Comment by dobkeratops** [comment]
i was just trying to make a suggestion list for 'scene labels' with default object labels (which the user could later confirm/deny... alternatively the system could determine them statistically with probabilities from whatever actual labels end up in the dataset ..
here's a quick list (it was just typed in excel so it's in CSV, 'scene label,label1,label2,,,,' etc
I'll try again in JSON later

```
village street,road,trees,bushes,pavement,buildings,person,car,,,
city street,road,trees,bushes,pavement,buildings,person,car,,,
town centre,road,trees,bushes,pavement,buildings,person,car,shop front,,
residential street,road,trees,bushes,pavement,buildings,person,car,shop front,house,apartment block
car park,car,barrier,pillar,fence,,,,,,
shopping mall,shop front,person,pillar,escalator,,,,,,
retail park,shop front,car park,car,person,building,pavement,road,,,
industrial area,building,person,,,,,,,,
rural area,trees,field,hedge,,,,,,,
airport,,,,,,,,,,
harbour,sea,boat,pier,buildings,,,,,,
coastline,sea,cliff face,mountain,,,,,,,
marina,boat,sea,pier,building,,,,,,
port,ship,gantry crane,pier/jetty,,,,,,,
docks,ship,gantry crane,container ship,shipping container,,,,,,
office,desk,computer monitor,office chair,man,woman,pen,,,,
park,grass,path,tree trunk,foliage,fence,,,,,
school,,,,,,,,,,
classroom,chair,chalkboard,whiteboard,pen,desk,child,book,,,
playground,,,,,,,,,,
hospital,bed,syringe,thermometer,,,,,,,
workshop,tool,workbench,person,,,,,,,
garage,car,tool,person,,,,,,,
home,,,,,,,,,,
kitchen,sink,hob,saucepan,bottle,plate,,,,,
bedroom,bed,lampshade,bedside cabinet,window,,,,,,
living room,TV,sofa,window,,,,,,,
dining room,dinner table,chair,plate,window,,,,,,
bathroom,sink,toilet,bath,shower head,,,,,,
hotel room,chair,desk,bed,,,,,,,
restaurant,table,chair,food,plate,sign,,,,,
pub/bar,table,chair,bottle,,,,,,,
cafe,table,chair,,,,,,,,
warehouse,,,,,,,,,,
supermarket,shelf,food,,,,,,,,
farm,field,tractor,hedge,livestock,fence,,,,,
landscape,mountain ,sky,lake,river,,,,,,
sports stadium,person,ball,stadium,crowd,sportsperson,,,,,
riverside,river,person,building,sky,,,,,,

```

**Comment by dobkeratops** [comment]
Another suggestion, having taken some more photos lately (upload soon), I wondered if you would consider a resolution override in the UI : keep the default of 1024 , but if you have a strong hint that the images are scenes with many objects, it might be useful to submit at a higher resolution (options like 1920, 2560,3840 wide to match various monitor formats ?)

**Comment by bbernhard** [comment]
Thanks for the labels list - that looks really interesting! 

I am wondering if we can somehow make it more transparent to the user which labels are part of a scene label? If I understood you correctly, then adding a scene label would automatically add a bunch of default object labels, right? 

In general I think this wouldn't be much of a problem as we do have the validation phase. But looking at the statistics it shows that only a small number of labeled objects is validated. Currently there are ~70k labeled objects and of those 70k labeled objects only ~15k are validated. Please don't get me wrong here...I don't wanna complain about those numbers here - 70k is totally awesome and is way above my expectations. This also shows that the labeling UI seems to be working pretty well at this point :)

On the other hand, those numbers also indicate that we maybe never reach a really good validation coverage. So I guess it becomes more and more important, that the initially picked label is correct, in order to maintain a good false-positive ratio. 

That's why my recent image donations were all unlabeled ones. It's just a gut feeling, but I think it's easier and more fun to label an image then to validate a labeled object. I recently scraped flickr for public domain images and while doing so, I realized that they have *a lot* of wrongly tagged images. I searched for the keywoard `flower` and nearly half of the images were not flowers at all. That's when I decided to upload all those images as unlabeled ones. 
 

> having taken some more photos lately (upload soon)

awesome, great to hear!

> I wondered if you would consider a resolution override in the UI : keep the default of 1024 , but if you have a strong hint that the images are scenes with many objects, it might be useful to submit at a higher resolution (options like 1920, 2560,3840 wide to match various monitor formats ?). The earlier scrape gives you a large image count, but if original contributions come through slower perhaps you can justify keeping more resolution

I did a few tests a few months ago and the problem with larger image sizes was, that it took significantly longer until the all the images in the browse based view were loaded. Unfortunately, it wasn't a network bottleneck (there's already server side image resizing in place in order to reduce traffic), but more a filesize issue. Due to larger image sizes, it takes significantly longer to read them from the disk...also the RAM usage during image resizing increases. I think the only way to circumvent that would probably be to save different sized thumbnails on the server. But this again increases the disk space needed :/

But I am planning to move the data from the digitalocean cloud to a dedicated root server (most probably a [Hetzner](https://www.hetzner.com) root server). As the server will be more powerful then, I guess it makes sense to re-evaluate those things :)

**Comment by dobkeratops** [comment]
>> then adding a scene label would automatically add a bunch of default object labels, right?

... it could . At the very least, in the "label objects" view, it could guide the label list that appears.

It may well be safe to have a series of object labels added automatically, and you could just remove (with 'skip' when asked?) the ones that aren't actually there (most street images will have car & person somewhere, but a few wont. almost all will have building+road+pavement+sky)

The most advanced implementation would be to measure a probability of each object label associated with a scene label. (eg highstreet would have 'shop front' very high,  a plain street would have 'shop front' lower).

> , but more a filesize issue. 

ok fair enough. Another idea was to crop high res objects from the interesting scenes. Those could be painted (with alpha channels?) for submission as seperate images. that might risk annotating the same object twice, conversely the 'zoom in' could be used for part annotation . 

If there was a way to flag a primary label for those, you could take the hint that a vision net run on that image should give the primary label as the strongest answer

```docks``` - whole image scaled to 1024

```boat``` - crops from original ('primary label=boat' for each crop?)
<img width="101" alt="Screenshot 2019-04-16 at 22 02 42" src="https://user-images.githubusercontent.com/1120754/56243907-94895d00-6093-11e9-9292-e885e719ab0f.png">
<img width="97" alt="Screenshot 2019-04-16 at 22 02 38" src="https://user-images.githubusercontent.com/1120754/56243914-97844d80-6093-11e9-9c70-4eff2b13b8d9.png">
<img width="546" alt="Screenshot 2019-04-16 at 22 02 54" src="https://user-images.githubusercontent.com/1120754/56243922-9ce19800-6093-11e9-82bc-7593587f62ac.png">
